implementation/drone/components/gun/src/java/org/inaetics/dronessimulator/drone/components/gun/Gun.java

```diff
@@ -0,0 +1,60 @@
+package org.inaetics.dronessimulator.drone.components.gun;
+
+import org.inaetics.dronessimulator.common.D3PoolCoordinate;
+import org.inaetics.dronessimulator.common.D3Vector;
+import org.inaetics.dronessimulator.common.protocol.*;
+import org.inaetics.dronessimulator.drone.components.gps.GPS;
+import org.inaetics.dronessimulator.drone.droneinit.DroneInit;
+import org.inaetics.dronessimulator.pubsub.api.Message;
+import org.inaetics.dronessimulator.pubsub.api.publisher.Publisher;
+import org.inaetics.dronessimulator.pubsub.api.subscriber.Subscriber;
+
+import java.io.IOException;
+import java.util.UUID;
+
+/**
+ * Created by mart on 17-5-17.
+ */
+public class Gun {
+    private volatile Subscriber m_subscriber;
+    private volatile Publisher m_publisher;
+    private volatile DroneInit m_drone;
+    private volatile GPS m_gps;
+    private final double GUN_SPEED = 50.0;
+    private final double MAX_DISTANCE = 100;
+
+
+    public double getMaxDistance(){
+        return MAX_DISTANCE;
+    }
+
+
+    D3Vector optimize_acceleration(D3Vector input, double max_length){
+        double correctionFactor = 0;
+        // Prevent that the acceleration exteeds te maximum acceleration
+        if(input.length() > max_length){
+            correctionFactor = max_length / input.length();
```

Comment:
Vector math not completely correct. Should have been max_length / input.length()